vllm/core/scheduler.py

```diff
@@ -281,72 +302,214 @@ def _schedule(self) -> SchedulerOutputs:
                     preempted.append(seq_group)
                     break
             else:
-                # Append new slots to the sequence group.
+                logger.debug(f"append slot for {seq_group}")
                 self._append_slot(seq_group, blocks_to_copy)
-                running.append(seq_group)
-        self.running = running
-
-        # Swap in the sequence groups in the SWAPPED state if possible.
+                decoding_seq_groups.append(seq_group)
+                logger.debug(f"scheduled r -> r {seq_group.request_id}")
+                token_budget.record_decoding_tokens(
+                    seq_group.num_seqs(status=SequenceStatus.RUNNING) *
+                    self.num_decoding_tokens_per_seq)
+
+        # If any sequence group is preempted, do not swap in any sequence group.
+        if preempted:
+            return SchedulerDecodeOutputs(token_budget, decoding_seq_groups,
+                                          len(preempted), blocks_to_swap_in,
+                                          blocks_to_swap_out, blocks_to_copy)
+
+        # Step 2: Swap in the sequence groups in the SWAPPED state if possible.
```

Comment:
Before scheduling swapped requests, we should schedule incomplete prefills?

Reply:
yeah it is in the scope of the next PR. This PR only focuses on refactoring (thus it is easier to write different ordering for chunked prefill). The order of scheduling in the current PR should be exactly the same as master.